packages/adp-tooling/src/base/validate.ts

```diff
@@ -45,19 +47,25 @@ export default class UI5Validator {
      * Checks if middleware has all the necessary properties an adaptation project needs.
      *
      * @param customConfiguration AdpCustomConfiguration
-     * @param previewMiddleware CustomMiddleware
+     * @param fioriPreview CustomMiddleware
+     * @param fioriProxy CustomMiddleware
      */
     private static checkMiddlewareProperties(
         customConfiguration: AdpCustomConfiguration | undefined,
-        previewMiddleware: CustomMiddleware<PreviewMiddlewareConfiguration> | undefined
+        fioriPreview: CustomMiddleware<FioriToolsPreviewConfig> | undefined,
+        fioriProxy: CustomMiddleware<FioriToolsProxyConfig> | undefined
     ): void {
-        if (!customConfiguration || !previewMiddleware) {
+        if (!customConfiguration || !fioriPreview || !fioriProxy) {
             throw new Error('Missing required custom middleware or custom configuration in ui5.yaml');
         }
         this.assertProperties(['environment'], customConfiguration);
```

Comment:
We have removed `environment` from customConfiguration. It is ABAP by default. We have added safeMode property to customConfiguration, this one probably needs to be checked.